okhttp-sse/src/main/java/okhttp3/internal/sse/ServerSentEventReader.java

```diff
@@ -0,0 +1,138 @@
+package okhttp3.internal.sse;
+
+import java.io.IOException;
+import javax.annotation.Nullable;
+import okio.BufferedSource;
+import okio.ByteString;
+
+public final class ServerSentEventReader {
+  private static final ByteString CRLF = ByteString.encodeUtf8("\r\n");
+
+  public interface Callback {
```

Comment:
Will you expand this to include open, closed, error, reconnect events?

Reply:
This is an internal callback like WebSocketReader's. It's only for reporting things that come across the stream. The user version of the callback will have those events though.